dataproc_jupyter_plugin/handlers.py

```diff
@@ -166,7 +177,7 @@ async def post(self):
             await async_run_gcloud_subcommand(f"config set project {project_id}")
             await async_run_gcloud_subcommand(f"config set dataproc/region {region}")
             clear_gcloud_cache()
-            configure_gateway_client_url(self.config, self.log)
+            configure_gateway_client_url(self.config, self.log, "")
```

Comment:
The use of "" here will cause the previously set gateway URL (which might have used the configured project number) to be replaced with whatever can be read from gcloud.

I think we need to use the configured value, which I *believe* we can get with:

```py
DataprocPluginConfig.instance(parent=self.parent).kernel_gateway_project_number
```